server/channels/api4/channel.go

```diff
@@ -836,6 +836,11 @@ func getDeletedChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), c.Params.TeamId, model.PermissionReadPublicChannel) {
```

Comment:
Trying to understand why you chose `model.PermissionReadPublicChannel` and not `PermissionListTeamChannels`. The earlier seems to be specific to a particular channel.

Reply:
Not sure, I probably wasn't.  Updated.